### PR TITLE
Solomon test

### PR DESCRIPTION
**Pull request by sakateka**

### CODE REVIEW

File: senders/solomon/solomon_test.go [hunk 1]
```diff
@@ -3,32 +3,186 @@ package solomon
...
+		}
+		runtime.Gosched()
+	}
+	assert.Equal(t, 0, len(j))
+}
+func TestRequest(t *testing.T) {
+	netAddr := "127.0.0.1:35313"
+
+	// for network timeout test
+	l, _ := net.Listen("tcp", netAddr)
```
Comment (noxiouz) [inline senders/solomon/solomon_test.go]: use a port provided by Linux and use l.Addr()

Comment (sakateka) [inline senders/solomon/solomon_test.go]: Its a blackhole for net.Dial.
With ctxhttp this test actually not need, but with httpclient it should cover case where net timeout ocured.

File: senders/solomon/solomon.go [hunk 2]
```diff
@@ -2,11 +2,10 @@ package solomon
 
 import (
 	"bytes"
+	"context"
```
Comment (noxiouz) [inline senders/solomon/solomon.go]: as far as I know ctxhttp returns errors from non-standard context pakage, doesn't it?

Comment (sakateka) [inline senders/solomon/solomon.go]: strange why the tests did not fall)

**Comment by noxiouz** [comment]
+1

**Pull request merged**